types/react-responsive/index.d.ts

```diff
@@ -1,65 +1,90 @@
-// Type definitions for react-responsive 1.1.3
+// Type definitions for react-responsive 3.0
 // Project: https://github.com/contra/react-responsive
 // Definitions by: Alexey Svetliakov <https://github.com/asvetliakov>
+//                 Alec Hill <https://github.com/alechill>
+//                 Xavier Gonzalez <https://github.com/xaviergonz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
-
-declare module "react-responsive" {
-    import * as React from "react";
-
-    namespace MediaQuery {
-        export interface MediaQueryProps {
-            query?: string;
-            // matchers
-            orientation?: "portrait" | "landscape";
-            scan?: "progressive" | "interlace";
-            aspectRatio?: string;
-            deviceAspectRatio?: string;
-            height?: number | string;
-            deviceHeight?: number | string;
-            width?: number | string;
-            deviceWidth?: number | string;
-            color?: boolean;
-            colorIndex?: boolean;
-            monochrome?: boolean;
-            resolution?: number | string;
-            // types
-            minAspectRatio?: string;
-            maxAspectRatio?: string;
-            minDeviceAspectRatio?: string;
-            maxDeviceAspectRatio?: string;
-            minHeight?: number | string;
-            maxHeight?: number | string;
-            minDeviceHeight?: number | string;
-            maxDeviceHeight?: number | string;
-            minDeviceWidth?: number | string;
-            maxDeviceWidth?: number | string;
-            minWidth?: number | string;
-            maxWidth?: number | string;
-            minColor?: number;
-            maxColor?: number;
-            minColorIndex?: number;
-            maxColorIndex?: number;
-            minMonochrome?: number;
-            maxMonochrome?: number;
-            minResolution?: number | string;
-            maxResolution?: number | string;
-            // types
-            all?: boolean;
-            grid?: boolean;
-            aural?: boolean;
-            braille?: boolean;
-            handheld?: boolean;
-            print?: boolean;
-            projection?: boolean;
-            screen?: boolean;
-            tty?: boolean;
-            tv?: boolean;
-            embossed?: boolean;
-        }
-    }
-
-    class MediaQuery extends React.Component<MediaQuery.MediaQueryProps, any> { }
-    export = MediaQuery;
+// TypeScript Version: 2.4
 
+import * as React from "react";
+
+export interface MediaQueryTypes {
+    all?: boolean;
+    grid?: boolean;
+    aural?: boolean;
+    braille?: boolean;
+    handheld?: boolean;
+    print?: boolean;
+    projection?: boolean;
+    screen?: boolean;
+    tty?: boolean;
+    tv?: boolean;
+    embossed?: boolean;
 }
+
+export type MediaQueryType = keyof MediaQueryTypes;
```

Comment:
why not just the keys themselves as strings and then remove the interface? I don't think it's used anywhere

Reply:
It is used to extend the queryable properties for `MediaQueryAllQueryable` => can be applied to component and to the standalone `toQuery` helper

Allows for `<MediaQuery screen={true} handheld={true}>Some handheld device content</MediaQuery>`